src/todt.c

```diff
@@ -896,7 +896,7 @@ dt_t **toDtElem(TypeSArray *tsa, dt_t **pdt, Expression *e)
         pdt = dtend(pdt);
         Type *tnext = tsa->next;
         Type *tbn = tnext->toBasetype();
-        while (tbn->ty == Tsarray && (!e || tbn != e->type->nextOf()))
+        while (tbn->ty == Tsarray && (!e || !tbn->equivalent(e->type->nextOf())))
```

Comment:
@9rnsr, I've tried this patch in 2.068 and it segfaults, because e->type->nextOf() is null. Maybe there should be a check for it ?